versions/3.0.4.md

```diff
@@ -96,6 +96,19 @@ Some examples of possible media type definitions:
 The HTTP Status Codes are used to indicate the status of the executed operation. 
 Status codes SHOULD be selected from the available status codes registered in the [IANA Status Code Registry](https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml).
 
+##### <a name="undefinedAndImplementationDefinedBehavior"></a>Undefined and Implementation-Defined Behavior
+
+This specification deems certain situations to have either _undefined_ or _implementation-defined_ behavior.
+
+Behavior described as _undefined_ is likely, at least in some circumstances, to contradict the specification.
+This description is used when detecting the contradiction is impossible or impractical.
+Implementations MAY support undefined scenarios for historical reasons, including ambiguous text in prior versions of the specification.
+This support might produce correct outcomes in many cases, but relying on it is NOT RECOMMENDED as there is no guarantee that it will work across all tools or with future specification versions, even if those versions are otherwise strictly compatible with this one.
+
+Behavior described as _implementation-defined_ allows implementations to choose which of several different-but-compliant approaches to a requirement to implement.
+This documents ambiguous requirements that API description authors are RECOMMENDED to avoid in order to maximize interoperability.
+Unlike undefined behavior, it is safe to rely on implementation-defined behavior if _and only if_ it can be guaranteed that all relevant tools support the same behavior.
```

Comment:
Well, that guarantee may be hard to come by 😁

Reply:
Indeed... it mirrors a few other spec passages, like the one in the JSON RFC that says you can use something other than UTF-8 only in a "closed system."

But it's not that uncommon to have internal APIs where your customers are mandated to use a particular client or technology stack.  It's not something I'd go out of my way to accommodate, but there _are_ scenarios where you can be pretty certain that you won't hit a different interpretation of an implementation-defined requirement.

If nothing else, some people don't do anything with OpenAPI but generate docs.